segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/ContainerKeyIndex.java

```diff
@@ -687,17 +698,10 @@ private void triggerCacheTailIndex(DirectSegmentAccess segment, long tailCaching
                 .processAll(rr, this.executor, this.config.getRecoveryTimeout())
                 .thenApplyAsync(inputData -> {
                     // Parse out all Table Keys and collect their latest offsets, as well as whether they were deleted.
-                    val updates = new TailUpdates();
                     collectEntriesWithHighestVersion(inputData, startOffset, maxLength, updates, tailCachePreIndexVersionTracker, lastIndexedOffset);
```

Comment:
Right now this method can silently fail which will look like a success to this code. It shouldn't have that try/catch

Reply:
If I understand the comment correctly, you refer to the try/catch inside `collectEntriesWithHighestVersion()` right? If so, I think that the exception handling there is right: the only exception that is being caught is `BufferView.Reader.OutOfBoundsException`, which is a normal case when the read boundary falls within an entry and we read a partial entry. In this case, that exact exception will be thrown and `TailUpdates` will keep as `maxOffset` whatever was the last complete entry read (because the `while` loop in `collectEntriesWithHighestVersion()` is interrupted). This exact offset is then used by the main `Futures.loop` by setting it as the start offset for the next batch iteration at `thenAccept(preIndexOffset::set)`. With this, the processing reads again the partial entry, this time correctly, and continues the processing. In fact, the test I have added does many combinations of batch sizes, as I was initially looking for a bug in that aspect, but it is handling partial entries correctly. Also, if any other exception occurs, it won't be caught and it will get to the `exceptionally()` block in `Futures.loop`, if I'm not mistaken.

Reply:
As discussed, lets add a log line and file an issue for the larger refactoring as it affects other classes.